src/features/postsSlice.ts

```diff
@@ -0,0 +1,52 @@
+/* eslint-disable no-param-reassign */
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { getUserPosts } from '../api/posts';
+import { Post } from '../types/Post';
+
+export interface PostsState {
+  loaded: boolean
+  hasError: boolean
+  items: Post[]
+}
+
+const initialState: PostsState = {
+  loaded: false,
+  hasError: false,
+  items: [],
+};
+
+export const loadPosts = createAsyncThunk(
+  'posts/set',
```

Comment:
Are you `posts/set` is a good name for loading user posts?

Reply:
I guess `posts/load` would be the better choice